changelog.d/12008.removal

```diff
@@ -0,0 +1 @@
+Remove support for the legacy structured logging configuration (using `structured: true` in the Synapse configuration).
```

Comment:
```suggestion
Remove support for the legacy structured logging configuration (i.e. using `structured: true` in the Synapse configuration).
```

Feels a bit like nitpicking, but I can see people reading this as "you should now use `structured: true`". Also could be worth linking to the upgrade notes.